pyiron_base/jobs/job/core.py

```diff
@@ -444,13 +490,8 @@ def remove_child(self):
 
         # Delete job from HPC-computing-queue if it is still running.
         job_status = str(self.status)
-        if (
-            job_status in ["submitted", "running", "collect"]
-            and "server" in self.project_hdf5.list_nodes()
-        ):
-            server_hdf_dict = self.project_hdf5["server"]
-            if "qid" in server_hdf_dict.keys() and server_hdf_dict["qid"] is not None:
-                self.project.queue_delete_job(server_hdf_dict["qid"])
+        if job_status in ["submitted", "running", "collect"]:
```

Comment:
Synchronize the job status checking with the kill method.  Here it's 'submitted', 'running', and 'collect' (which makes sense imo), but the kill method doesn't check for 'collect'.